### PR TITLE
Add new exception type: MessageIsNotModifiedException

### PR DESCRIPTION
**Pull request by karb0f0s**
Exception for "Bad Request: message is not modified" error
[Issue #596](https://github.com/TelegramBots/Telegram.Bot/issues/596)

### CODE REVIEW

File: src/Telegram.Bot/Exceptions/BadRequestExceptions/MessageIsNotModifiedException.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// ReSharper disable once CheckNamespace
+namespace Telegram.Bot.Exceptions
+{
+    /// <summary>
+    /// The exception that is thrown when the sticker set is invalid
```
Comment (pouladpld) [inline src/Telegram.Bot/Exceptions/BadRequestExceptions/MessageIsNotModifiedException.cs]: copy-pasta :)

File: test/Telegram.Bot.Tests.Integ/Exceptions/ApiExceptionsTests.cs [hunk 2]
```diff
@@ -103,6 +124,9 @@ private static class FactTitles
             public const string ShouldThrowExceptionContactRequestException =
                 "Should throw ContactRequestException while asking for user's phone number in non-private " +
                 "chat via reply keyboard markup";
+
+            public const string ShouldThrowExceptionMessageIsNotModifiedException =
+               "Should throw MessageIsNotModifiedException while editing previously sent message";
```
Comment (pouladpld) [inline test/Telegram.Bot.Tests.Integ/Exceptions/ApiExceptionsTests.cs]: add
_... previously sent message with the same text_

File: test/Telegram.Bot.Tests.Integ/Exceptions/ApiExceptionsTests.cs [hunk 3]
```diff
@@ -88,6 +88,27 @@ public ApiExceptionsTests(TestsFixture fixture)
...
 
+        [Fact(DisplayName = FactTitles.ShouldThrowExceptionMessageIsNotModifiedException)]
+        [Trait(Constants.MethodTraitName, Constants.TelegramBotApiMethods.SendMessage)]
+        [ExecutionOrder(5)]
+        public async Task Should_Throw_Exception_MessageIsNotModifiedException()
+        {
+            await _fixture.SendTestCaseNotificationAsync(FactTitles.ShouldThrowExceptionMessageIsNotModifiedException);
+
+            const string MessageTextToModify = "Message text to modify";
+            var message = await BotClient.SendTextMessageAsync(
```
Comment (pouladpld) [inline test/Telegram.Bot.Tests.Integ/Exceptions/ApiExceptionsTests.cs]: no `var` in tests. Write the type name. This helps people to reason about the code by just reading it.

**Comment by karb0f0s** [comment]
I made requested corrections

**Pull request merged**